### DISCUSSION TITLE
Slow TypeScript Language Server Performance with Panda CSS

### DISCUSSION THREAD

**Discussion by evvarela** [chakra-ui/panda#2977] [General]
### **Description**
When using Panda CSS, the TypeScript Language Server experiences significant performance issues. Deleting the `styled-system` folder generated by Panda CSS temporarily resolves the issue, suggesting the folder contents may be contributing to the slowdown.

### **Steps to Reproduce**
1. Install and set up Panda CSS in the project.
2. Start the TypeScript Language Server in an editor (e.g., VSCode).
3. Observe slow performance in TypeScript-related tasks (e.g., IntelliSense, type-checking).
4. Delete the `styled-system` folder generated by Panda CSS.
5. Notice that TypeScript Language Server performance returns to normal.

### **Expected Behavior**
The TypeScript Language Server should perform smoothly.

### **Actual Behavior**
The TypeScript Language Server becomes noticeably slow when the `styled-system` folder is present, impacting productivity and developer experience.

### **Additional Context**
Any insights into why the `styled-system` folder might be causing this issue or potential workarounds would be appreciated.

```{
  "include": [
    "**/*.ts",
    "**/*.tsx",
    "**/.server/**/*.ts",
    "**/.server/**/*.tsx",
    "**/.client/**/*.ts",
    "**/.client/**/*.tsx",
    ".react-router/types/**/*",
    "../shared/**/*",
    "styled-system",
  ],
  "exclude": ["node_modules", "build", "../server/**", "tmp"
[Screencast from 11-12-2024 04:38:18 PM.webm](https://github.com/user-attachments/assets/139b8f19-632c-4278-a30f-54d1bae48f88)
],
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "ES2022"],
    "types": ["@react-router/node", "vite/client", "vite-plugin-svgr/client"],
    "isolatedModules": true,
    "esModuleInterop": true,
    "jsx": "react-jsx",
    "module": "ESNext",
    "incremental": true,
    "tsBuildInfoFile": "./.tsbuildinfo",
    "sourceMap": true,
    "moduleResolution": "Bundler",
    "resolveJsonModule": true,
    "target": "ES2022",
    "strict": true,
    "allowJs": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "baseUrl": ".",
    "paths": {
      "~/*": ["./app/*"],
      "@shared/*": ["../shared/*"]
    },
    "noEmit": true,
    "rootDirs": [".", "./.react-router/types"],
    "plugins": [{ "name": "@react-router/dev" }]
  }
}
```

### Link to Reproduction

https://stackblitz.com/edit/vitejs-vite-lfwyue?file=src%2FApp.tsx&terminal=dev

### JS Framework

React (TS)

### Panda CSS Version

0.47.1

### Operating System

- [ ] macOS
- [ ] Windows
- [X] Linux

**Comment by evvarela** [comment]
[Screencast from 11-12-2024 04:38:18 PM.webm](https://github.com/user-attachments/assets/4d5c0f4a-6813-4a20-a225-09b111396ec8)

**Reply by cgatian** [reply]
Issues like this are typically due to an extension.
